lib/modules/manager/terraform/lockfile/index.ts

```diff
@@ -191,6 +187,10 @@ export async function updateArtifacts({
             continue;
           }
         }
+
+        // use registryURL defined in the update and fall back to the one defined in the lockfile
+        const registryUrl = registryUrls?.[0] ?? updateLock.registryUrl;
```

Comment:
we should read it on extract, so no user intervention is required .

the datasource is always filling registry url, so you fallback will probably never used at runtime

Reply:
I agree we should detect Terraform vs OpenTofu during extract, though this is not possible if there is no lock file. At least I'm not aware there is a reliable way. 

This is picked up as expected, I have tested / debugged it against a real repo.